### DISCUSSION TITLE
Initialization file WFS example.

### DISCUSSION THREAD

**Discussion by workmap** [TerriaJS/terriajs#5365] [Q&A]
Hi, I'm trying to add a WFS to my initialization file. I am adding it in like this:

`{

                  "name": "WFS Example",

                  "type": "wfs",

                  "layers": "test:exampleLayer",

                  "url": "https://example.com/geoserver/wfs/test?service=wfs&version=1.3.0&request=GetCapabilities"

                }`

I can add the layer to Terriamap without error, however none of the features show up.  Anyone have an example of how they add WFS layers to their configuration?  

Interestingly, when I set the `type` to `wfs-getCapabilities` the WFS layer loads but within a group. I'd prefer that my WFS not be loaded into a group folder, however. What property is needed to load the WFS not as a group but as a layer (e.g. `type: 'wfs'`)?

**Comment by lstillwe** [comment]
I would also like to know the answer to this question.

Thanks!

**Reply by workmap** [reply]
use typeNames rather than layers like this:

`{
    "name": "WFS Example",

    "type": "wfs",

    "typeNames": "test:exampleLayer",

    "url": "https://example.com/geoserver/wfs/test?service=wfs&version=1.3.0&request=GetCapabilities"
}`

**Reply by lstillwe** [reply]
Thank you! Do you know if one can then use a featureInfoTemplate in this example?
IE exampleLayer has attributes photo_name and photo_url

> \"featureInfoTemplate": {
>             “template": "\<div class=’photos’>
>                 \<figure>\<img src={{photo_url}}>
>                   \<figcaption>{{photo_name}}>
>                   \</figcaption>
>                  \</figure>\</div>"
> \}
I have not been able to get this to work yet.

**Reply by lstillwe** [reply]
The answer to my own followup question is yes. You can use a featureInfoTemplate.
I could not get my previous example to work, but this finally did work:

> \"featureInfoTemplate":  "\<div class=’photos’>
>                 \<figure>\<img src={{photo_url}}>
>                   \<figcaption>{{photo_name}}>
>                   \</figcaption>
>                  \</figure>
> \</div>"
> \}
